network/src/validator_network/network_builder.rs

```diff
@@ -455,48 +484,59 @@ impl NetworkBuilder {
     /// Create the configured transport and start PeerManager.
     /// Return the actual NetworkAddress over which this peer is listening.
     pub fn build(mut self) -> NetworkAddress {
+        use libra_network_address::Protocol::*;
+
         let peer_id = self.peer_id;
-        let supported_protocols = self.supported_protocols();
-        // Build network based on the transport type
+        let protos = self.supported_protocols();
+
         let trusted_peers = self.trusted_peers.clone();
-        match self.transport {
-            TransportType::Memory => {
-                self.build_with_transport(build_memory_transport(peer_id, supported_protocols))
-            }
-            TransportType::MemoryNoise(ref mut keys) => {
-                let keys = keys.take().expect("Identity keys not set");
-                self.build_with_transport(build_memory_noise_transport(
-                    keys,
-                    trusted_peers,
-                    supported_protocols,
-                ))
-            }
-            TransportType::PermissionlessMemoryNoise(ref mut keys) => {
-                let keys = keys.take().expect("Identity keys not set");
-                self.build_with_transport(build_unauthenticated_memory_noise_transport(
-                    keys,
-                    supported_protocols,
-                ))
-            }
-            TransportType::Tcp => {
-                self.build_with_transport(build_tcp_transport(peer_id, supported_protocols))
+        let auth_mode = self.auth_mode.take().expect("Authentication Mode not set");
```

Comment:
random idea: we could assert somewhere that the auth_mode is mutual on validator network, server side on peer network, and unauthenticated only if the config is marked as "test" (we could have the first field of the config be something set to production or test)

Reply:
yeah, I think the configs as a whole need some broader refactoring work :'(

also to help reduce some of these invalid states